app/controllers/account_activations_controller.rb

```diff
@@ -10,13 +10,19 @@ def show
   
   def edit
     user = User.find_by(email: params[:email].downcase)
-    if user && !user.activated? && @user.authenticated?(:activation, params[:id])
+    if user && !user.activated? && @user.authenticated?(:activation, activation_token)
```

Comment:
If statements like this are always a little bit icky. You're also mixing `user` and `@user` and I'm not sure which one it should be.

I would use `@user = User.find_by(email: params[:email].downcase)` because it's expected that the primary variable of a controller is an instance variable, even if you're not really using it outside of this action.

The `if` statement can also be simplified, for instance by returning early if a user can not be found:

```ruby
@user = User.find_by(email: params[:email].downcase)
unless @user.present?
  # ... Set flash and redirect because no user
  return false
end

user_activated = user.activated? || @user.activate(activation_token)
if user_activated
  # ...
else
  # ...
end
```

You would have to modify the `activate` method in the `User` model to accept the activation_token and check it there instead of in the `authenticated?` method, which feels a little bit weird anyway.